modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/wstep/impl/CertificateEnrollmentServiceImpl.java

```diff
@@ -0,0 +1,386 @@
+import org.wso2.carbon.mdm.mobileservices.windows.services.wstep.CertificateEnrollmentService;
+import org.wso2.carbon.mdm.mobileservices.windows.services.wstep.beans.BinarySecurityToken;
+import org.apache.commons.codec.binary.Base64;
+import javax.annotation.Resource;
+import javax.jws.WebService;
+import javax.servlet.ServletContext;
+import javax.xml.bind.DatatypeConverter;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.ws.BindingType;
+import javax.xml.ws.Holder;
+import javax.xml.ws.WebServiceContext;
+import javax.xml.ws.handler.MessageContext;
+import javax.xml.ws.soap.Addressing;
+import javax.xml.ws.soap.SOAPBinding;
+
+import org.wso2.carbon.mdm.mobileservices.windows.services.wstep.util.CertificateSigningService;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.wso2.carbon.mdm.mobileservices.windows.services.wstep.beans.RequestSecurityTokenResponse;
+import org.wso2.carbon.mdm.mobileservices.windows.services.wstep.beans.RequestedSecurityToken;
+import org.wso2.carbon.mdm.mobileservices.windows.services.wstep.util.KeyStoreGenerator;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.bouncycastle.pkcs.PKCS10CertificationRequest;
+import org.bouncycastle.pkcs.jcajce.JcaPKCS10CertificationRequest;
+import org.w3c.dom.Document;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+import org.wso2.carbon.mdm.mobileservices.windows.common.beans.WindowsPluginProperties;
+
+/**
+ * Implementation class of CertificateEnrollmentService interface. This class implements MS-WSTEP
+ * protocol.
+ */
+@WebService(endpointInterface = Constants.CERTIFICATE_ENROLLMENT_SERVICE_ENDPOINT,
+		targetNamespace = Constants.DEVICE_ENROLLMENT_SERVICE_TARGET_NAMESPACE)
+@Addressing(enabled = true, required = true)
+@BindingType(value = SOAPBinding.SOAP12HTTP_BINDING)
+public class CertificateEnrollmentServiceImpl implements CertificateEnrollmentService {
+
+	public static final int FIRST_ITEM = 0;
```

Comment:
The constant name "FIRST_ITEM" doesn't mean anything. Use more contextual information within the name itself.